### ISSUE TITLE
plan to opensource code about how to generate the training data? thanks

### ISSUE THREAD

**Issue by ericg108** [Agent-RL/ReSearch#1]
Hi there,

it's good to see another work to integrate reasoning with search. 

Do you guys have plans to share the code about how to generate the training data, which will make the whole work easier to reproduce?

Thanks

**Comment by AnselCmy** [comment]
Thank you for reaching out! For detailed information, please refer to this section in the README: [Train the LLM via VERL](https://github.com/Agent-RL/ReSearch?tab=readme-ov-file#train-the-llm-via-verl). We utilize the open HotpotQA dataset as our training data, and no additional data preparation steps are required. If you have any more questions, feel free to ask!

**Comment by ericg108** [comment]
thank you for your answer. I thought the training dataset is generated before training seperately. 
Now from the `run.sh`, it seems the dataset is generated on the fly, the code of `main_ppo.py` will do that, right?

**Comment by AnselCmy** [comment]
As mentioned, You can download the HotpotQA dataset from [here](https://hotpotqa.github.io/). The data preprocess script is provided in training/data_preprocess_hpqa.py, and it will generate the train and dev data in parquet format.

**Comment by ericg108** [comment]
Thank you for your answer. Please excuse my inexperience.

yes. I got the dataset. 
I mean, the HotpotQA dataset from [here](https://hotpotqa.github.io/) is the original dataset and the training/data_preprocess_hpqa.py is just splitting the train/dev with appending a reward_model field in it. 
so until here, the hotpotqa dataset is processed into a parquet format but without the "thinking" part, which, I guess, should contain when to search and when to answer, something like that, so that the model can learn to search at the right time.

But I dont see where the "thinking" part is generated. so I guess it's done in the main_ppo.py?
I'm confused about that, thanks.

**Comment by AnselCmy** [comment]
This is just the amazing part of the reinforcement leaning, there is no need to collect any 'thinking' data for model to imitate that. Only use simple reward signal, model can 'learn how to think' by itself. You can also refer to the similar setting from the paper of deepseek-r1.

**Comment by ericg108** [comment]
wow, that's amazing indeed. I have a lot to catch up. thanks.
